algorithms/linfa-linear/src/ols.rs

```diff
@@ -205,10 +204,13 @@ where
     B: Data<Elem = F>,
     C: Data<Elem = F>,
 {
-    let rhs = X.t().dot(y);
-    let linear_operator = X.t().dot(X);
-    linear_operator
-        .solve_into(rhs)
+    // It would be cleaner to use ndarray_linalg::LeastSquaresSvd::least_squares,
+    // but that is currently not possible since B = C is required
+    let X = X.to_owned();
```

Comment:
can you accept a view representation as parameter and avoid the copy here?

Reply:
I don't see how, since the types of the data containers for `X` and `y` are different (`B` and `C` respectively). In an upcoming version of `ndarray-linalg` this restriction has been lifted so we don't have to make an explicit copy here (https://github.com/rust-ndarray/ndarray-linalg/pull/272).

Reply:
the data is copied for the intercept branch anyways. I pushed a PR https://github.com/janmarthedal/linfa/pull/1 which adds the copy only for the branch without intercept computation

Reply:
Ah, I see, that makes sense. Thank you for helping out!